lib/LocalizationProvider.js

```diff
@@ -90,7 +109,26 @@ LocalizationProvider.prototype._pluralizationRulesCache = null;
  * @param {Object} context Component context.
  */
 LocalizationProvider.prototype.getCurrentLocale = function (context) {
-	return context.cookie.get(this._cookieName) || this._defaultLocale;
+	return context.cookie.get(this._cookieConfig.name) || this._defaultLocale;
+};
+
+/**
+ * Changes current locale value.
+ * @param {string} locale Locale name (i.e. en, en-us, ru etc).
+ * @param {Object} context Component context.
+ */
+LocalizationProvider.prototype.changeLocale = function (locale, context) {
+	if(context.isBrowser) {
+		var window = context.locator.resolve('window'),
+			expireDate = new Date((new Date()).getTime() +
+				this._cookieConfig.maxAge * 1000);
+
+		this._cookieConfig.key = this._cookieConfig.name;
+		this._cookieConfig.value = locale;
+		this._cookieConfig.expires = expireDate.toUTCString();
+		context.cookie.set(this._cookieConfig);
+		window.document.location.reload();
```

Comment:
I meant you should put the 
```javascript
var window = context.locator.resolve('window');
window.document.location.reload();
``` 
expressions to `if(context.isBrowser)` only. 
All other `cookie` stuff will work properly on the server environment too.

All other stuff is great.